examples/simple/NeighborListCollector.h

```diff
@@ -0,0 +1,32 @@
+#ifndef PARATREET_NEIGHBORLISTCOLLECTOR_H_
+#define PARATREET_NEIGHBORLISTCOLLECTOR_H_
+
+#include "common.h"
+#include "Vector3D.h"
+#include "paratreet.decl.h"
+
+#include <cstring>
+#include <cmath>
+
+struct NeighborListCollector : public CBase_NeighborListCollector {
+  using Neighborhood = std::vector<std::pair<Key, Vector3D<Real>>>;
+  std::map<Key, Neighborhood> local_neighbors;
+  std::map<Key, Neighborhood> remote_neighbors;
+
+  void share(const CkCallback& cb) {
+    // TODO copy local neighbors to remote neighbors. but how to tell which PE to send?
```

Comment:
@Wingpad any ideas here? might add a field to each Node that holds the Subtree's Rank instead of the cm_index and use the Rank to get the cm_index. And then i can just copy all the neighborhoods over in one big message. thoughts?

Reply:
Of course, I could use the stuff in Decomposition to "flush" the forces, but that seems excessive

Reply:
What is the formula for the destination PE?

Reply:
It's just the location of the source particle, which can be found using the Decomposition

Reply:
So, would adding something like `locationOf(Key)` to `Decomposition` solve this issue? It's globally accessible, and "local" so anything in the system should be able to query it.

Reply:
Used home_pe to solve this problem.